src/main/java/io/odpf/firehose/sink/AbstractSink.java

```diff
@@ -26,30 +31,56 @@
      * Method to push messages to sink.
      *
      * @param messages the messages
-     * @return the list
+     * @return the list of failed messages
      * @throws DeserializerException when invalid kafka message is encountered
      */
     public List<Message> pushMessage(List<Message> messages) throws DeserializerException {
-        List<Message> failedMessages;
+        List<Message> failedMessages = messages;
         try {
             instrumentation.logDebug("Preparing {} messages", messages.size());
+            instrumentation.captureMessageBatchSize(messages.size());
+            instrumentation.captureMessageMetrics(Metrics.SINK_MESSAGES_TOTAL, Metrics.MessageType.TOTAL, messages.size());
```

Comment:
are message metrics solving anything better, if we have have global metrics that should be sufficient. I believe we might not  need this metric

Reply:
Global message metric counts one message only one time... This metric tells how many messages were processed by SINK including retries.